pkg/apis/clusterpedia/v1beta1/conversion.go

```diff
@@ -229,6 +252,45 @@ func convert_Slice_string_To_String(in *[]string, out *string, scope conversion.
 	return nil
 }
 
+func convert_String_To_Pointer_time(in *string, out **metav1.Time, scope conversion.Scope) error {
+	if strings.Contains(*in, "-") {
+		*in = strings.ReplaceAll(*in, "Z", "+")
+		t, err := time.Parse(time.RFC3339, *in)
+		if err == nil {
+			*out = &metav1.Time{Time: t}
+			return nil
+		}
+		t, err = time.Parse("2006-01-02", *in)
+		if err == nil {
+			*out = &metav1.Time{Time: t}
+			return nil
+		}
+		t, err = time.Parse("2006-01-02 15:04:05", *in)
+		if err == nil {
+			*out = &metav1.Time{Time: t}
+			return nil
+		}
+	} else {
+		timestamp, err := strconv.ParseInt(*in, 10, 64)
+		if err != nil {
+			return err
+		}
+		if len(*in) == 10 {
+			t := time.Unix(timestamp, 0)
+			*out = &metav1.Time{Time: t}
+			return nil
+		} else if len(*in) == 13 {
+			t := time.UnixMilli(timestamp)
+			*out = &metav1.Time{Time: t}
+			return nil
+		}
```

Comment:
Distinguishing timestamp units by the length of the timestamp string is indeed a more user-friendly way of doing things, but there may be some unusual behaviour when used incorrectly by users, e.g. since is 10 and before is 13.

We need to determine whether before is greater than since.